app/services/remove_status_service.rb

```diff
@@ -57,14 +53,34 @@ def remove_reblogs
     end
   end
 
-  def unpush(type, receiver, status)
-    if status.reblog? && !redis.zscore(FeedManager.instance.key(type, receiver.id), status.reblog_of_id).nil?
-      redis.zadd(FeedManager.instance.key(type, receiver.id), status.reblog_of_id, status.reblog_of_id)
+  def unpush_bulk(type, receiver_ids, status)
+    return if receiver_ids.empty?
+
+    if status.reblog?
+      scores = redis.pipelined do
+        receiver_ids.each do |receiver_id|
+          redis.zscore(FeedManager.instance.key(type, receiver_id), status.reblog_of_id)
+        end
+      end
+
+      redis.pipelined do
+        scores.each_with_index do |score, index|
+          next if score.nil?
+          redis.zadd(FeedManager.instance.key(type, receiver_ids[index]), status.reblog_of_id, status.reblog_of_id)
+        end
+
+        receiver_ids.each do |receiver_id|
+          redis.publish("timeline:#{receiver_id}", @payload)
```

Comment:
That is not possible because the responses are not available in the pipeline. That is also the cause of the regression fixed by #3704.
See https://redis.io/topics/pipelining. It also suggests to write a Lua embedded script, but I think that is too much.